common/dubbo_dir_refresh.go

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package common
+
+import "sync"
+
+// DirMap is used by dubbo-getty, it can notify delete event to dubbo directory directly by getty session
+// look up at:
+// remoting/getty/listener.go
+// registry/directory/directory.go
+var DirMap sync.Map
```

Comment:
Why use public variable in this module? Although you have comment in this variable，but It can be used in all modules in dubbo-go.Even the module out of dubbo-go.

Reply:
I think it's the best way to achieve the direct notify from remoting.session to directory

Reply:
I can't find another way to link this two pkg. As there may be unexpected loop refer.

Reply:
请允许我使用中文，关于这个问题，我先说一下我的看法。
先看一下现在的结构，我没理解错应该是：
invoker-》client-》getty pool-》session（connection）
现在是直接加在 session 的 close/error 上。

针对这个 PR，我有以下几个疑问：
* 这个功能是否应该加在 invoker / client ？为什么我这么说？请看第二个问题。
* 因为有可能有 10 个 connection，其中 n （ n < 10 ）个因为网络抖动连不上，然后就去删除 DIr 中的 cache？这也可能和第三个问题有关联。
* 我只看到在 session（connection）上删除，但是我没看到如何加回去？
* 如果按照现在的套路，如果 n ( n > 1) 个 registry 要加回去，是不是会造成：频繁增加。

我的建议是：
调整到 invoekr 或者 client，通过判断窗口期中多少个请求失败，再触发删除 cache，最后记得加上恢复 registry 的代码。
但是如果按照我的建议做的话，是否能基于：ActiveFilter？

Reply:
我仔细想了想，这里的确有问题。
我写这个pr的是以：“这个session一旦被关闭，证明server端就一定关闭了”为前提的。一旦session因为网络抖动出现了err，会直接删除掉dir的cache，这样设计不太合理。
我先表述一下这个pr的初衷：是为了提高客户端对于server端下线的感知速度：不只通过regisgtry感知，而是通过session感知。如果判断窗口期请求，我觉得是不太现实，因为这个时间可能registry都通知过来了。
感觉这个需求可能有点问题。
另一方面，我的另一个pr：https://github.com/apache/dubbo-go/pull/976
有提供invoke前的健康检查机制，感觉这两个pr在功能上是有重复的，我和展图讨论下。